src/typing.py

```diff
@@ -1568,7 +1546,7 @@ def no_type_check(arg):
     if isinstance(arg, type):
         arg_attrs = arg.__dict__.copy()
         for attr, val in arg.__dict__.items():
-            if val in arg.__bases__:
+            if val in arg.__bases__ + (arg,):
```

Comment:
Yes, it is better to have them in both versions, ``@no_type_chek`` can be also used in Python 2, fixed this.